src/object/srv_obj.c

```diff
@@ -3464,9 +3464,14 @@ ds_obj_dtx_handle_one(crt_rpc_t *rpc, struct daos_cpd_sub_head *dcsh,
 			if (rc != 0)
 				goto out;
 		} else {
-			if (dcsr->dcsr_opc != DCSO_PUNCH_OBJ &&
-			    dcsr->dcsr_opc != DCSO_PUNCH_DKEY &&
-			    dcsr->dcsr_opc != DCSO_PUNCH_AKEY) {
+			daos_key_t	*dkey;
+
+			if (dcsr->dcsr_opc == DCSO_PUNCH_OBJ) {
+				dkey = NULL;
+			} else if (dcsr->dcsr_opc == DCSO_PUNCH_DKEY ||
+				   dcsr->dcsr_opc == DCSO_PUNCH_AKEY) {
+				dkey = &dcsr->dcsr_dkey;
+			} else {
```

Comment:
Should you set rc and add a skip or something here??   The dkey will be uninitialized garbage if this happens.   In fact, I wouldn't be surprised if my Fedora compiler complains with somtimes-uninitialized warning.

Reply:
I think we did.

                        } else {         
                                D_ERROR("Unknown sub request opc %u for obj "
                                        DF_UOID", DTX "DF_DTI":\n",
                                        dcsr->dcsr_opc, DP_UOID(dcsr->dcsr_oid),
                                        DP_DTI(&dcsh->dcsh_xid));
                
                                D_GOTO(out, rc = -DER_PROTO);
                        }

Reply:
ah, sorry, an artifact of github collapsing lines in the diff by default.  I really wish it had the gerrit setting where you could view a whole file instead without clicking to expand 500 times.   Sorry for the noise.